proposals/NNNN-input-semantics.md

```diff
@@ -0,0 +1,245 @@
+# Input semantics
+
+* Proposal: [NNNN](http://NNNN-input-semantics.md)
+* Author(s): [Nathan Gauër](https://github.com/Keenuts)
+* Status: **Design In Progress**
+
+## Introduction
+
+HLSL shaders can read form the pipeline state using [semantics](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-semantics).
+This proposal looks into how to implement input semantics in Clang.
+Output semantics are out of scope of this proposal, but some parts will be
+similar.
```

Comment:
I don't think input semantics should be separated from output semantics here.

A better distinction would be between all the processing and diagnostics that we want to perform before clang code gen (in Sema), and how to implement shader inputs and outputs in the IR during or after clang codegen.